src/librustc_typeck/astconv.rs

```diff
@@ -401,7 +401,17 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
                 0,
             );
         }
-        let ret = if !infer_types
```

Comment:
This will change the order of errors, which might be misleading as they'll no longer appear in the same order they are written. Probably not an issue, but worth noting in a comment at least.